pandas/tests/io/json/test_pandas.py

```diff
@@ -1032,6 +1032,67 @@ def test_to_jsonl(self):
         assert result == expected
         assert_frame_equal(pd.read_json(result, lines=True), df)
 
+    def test_read_jsonchunks(self):
+        # GH17048: memory usage when lines=True
+        df = pd.DataFrame({'A': [1, 2, 3], 'B': [4, 5, 6]})
+
+        def get_strio():
+            return StringIO(df.to_json(lines=True, orient="records"))
+
```

Comment:
this is also hard to follow, pls use some comments to say what you are testing

Reply:
Done